pydm/tests/data_plugins/test_p4p_plugin_component.py

```diff
@@ -0,0 +1,77 @@
+import functools
+from p4p.client.thread import Context
+from p4p.nt import NTScalar
+from pydm.data_plugins.epics_plugins.p4p_plugin_component import Connection, P4PPlugin
+from pydm.tests.conftest import ConnectionSignals
+from pydm.widgets.channel import PyDMChannel
+from pytest import MonkeyPatch
+
+
+class MockContext:
+    """ A do-nothing mock of a p4p context object """
+    def __init__(self):
+        self.monitor = None
+
+
+def test_send_new_value(monkeypatch: MonkeyPatch, signals: ConnectionSignals):
```

Comment:
Coverage for other data types + put functionality might be a good idea

Reply:
Added tests for additional types 69b31d058369fd2047b28c3e02c225d92728a15c